libcxx/test/std/utilities/expected/expected.expected/observers/has_value_clobber.pass.cpp

```diff
@@ -0,0 +1,34 @@
+//===----------------------------------------------------------------------===//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
```

Comment:
Should we also be adding a test for `expected<void>`?

Reply:
done! I added some tests that reproduce the issue more easily and without need for optimization.

Reply:
I just noticed there is a folder for `expected.void` tests so I moved the new `expected<void>` test there in commit https://github.com/llvm/llvm-project/pull/68733/commits/dfa9b3cf6e5951b4935e3e310b3e338f721beada.